src/include/txn.i

```diff
@@ -896,6 +898,34 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_UPDATE *upd, WT_UPDATE **updp)
 			skipped_birthmark = true;
 	}
 
+	/*
+	 * If there is a related lookaside entry, find it and try matching the
+	 * in-memory record. The following conditions will be refined in the
+	 * future once we make the change to stop initializing a page using
+	 * lookaside.
+	 */
+	if (upd != NULL && upd->txnid != 0 && cbt->ref->page_las != NULL &&
+	    __wt_find_lookaside_upd(session, cbt, &tmp_upd, true) == 0) {
+		/* It is okay to locate an older update in lookaside */
+		WT_ASSERT(session, upd->start_ts >= tmp_upd->start_ts);
+
+		if (upd->start_ts == tmp_upd->start_ts) {
```

Comment:
Why not pass the `start_ts` into `__wt_find_lookaside_upd()`? It felt a little odd to check lots of things in that function, but make that test in the caller instead?

Reply:
I was trying to avoid passing anything into ```__wt_find_lookaside_upd()``` from an in-memory update because the next ticket will do away with initializing updates from lookaside into memory. On second thought, this simplifies the code a lot. Also these asserts and conditions will go away when I do the next ticket anyways.

I have fixed this as you suggested, looks much cleaner.